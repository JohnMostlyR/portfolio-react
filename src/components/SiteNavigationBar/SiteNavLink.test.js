import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import {BrowserRouter as Router} from 'react-router-dom';
import {Link} from 'react-router-dom';
import { mount } from 'enzyme';

import SiteNavLink from './SiteNavLink';

describe('SiteNavLink', () => {
  const TO = '/test';
  const context = {
    router: {
      route: {
        location: {
          pathname: '/other',
        }
      }
    }
  };

  let wrapper;

  beforeEach(() => {
    wrapper = mount(
        <Router>
          <SiteNavLink to={TO} data-isactive={context.router.route.location.pathname === TO}/>
        </Router>
    );
  });

  it('Should render and match the snapshot', () => {
    const tree = renderer.create(<Router><SiteNavLink to={'/'} data-isactive={'/'}/></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should have a `Link` component', () => {
    expect(
        wrapper.find(Link)
    ).toHaveLength(1);
  });

  it('Should have an <a> where the href prop. matches the Link\'s to prop', () => {
    expect(
        wrapper.find(`a[href="${TO}"]`)
    ).toHaveLength(1);
  });

  it('Should have an <a> with a data-isactive prop. that is false when then browser' +
      ' location does not match the Link\'s to prop', () => {
    expect(
        wrapper.find(`a[href="${TO}"]`).prop('data-isactive')
    ).toBe(false);
  });
});