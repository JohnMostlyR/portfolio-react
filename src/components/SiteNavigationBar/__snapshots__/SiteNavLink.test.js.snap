// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteNavLink Should render and match the snapshot 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  background-color: #FF6633;
  border-radius: 0;
  color: #fff;
  cursor: pointer;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 0.5em;
  text-align: center;
  text-decoration: none !important;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  font-family: "Segoe UI",Roboto,"Droid Sans","Helvetica Neue",Helvetica,Arial,freesans,sans-serif;
  font-weight: 400;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  background-color: transparent;
  color: #FF6633;
}

.c0:active,
.c0:hover {
  background-color: transparent;
  color: #FF6633;
}

.c0:active::before,
.c0:hover::before {
  opacity: 1;
  -webkit-transform: scale3d(1,1,1);
  -ms-transform: scale3d(1,1,1);
  transform: scale3d(1,1,1);
}

.c0::before {
  content: '';
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 1;
  border: 1px solid #FF6633;
  border-radius: inherit;
  opacity: 0;
  -webkit-transform: scale3d(0.6,0.6,1);
  -ms-transform: scale3d(0.6,0.6,1);
  transform: scale3d(0.6,0.6,1);
  -webkit-transition: -webkit-transform 0.3s,opacity 0.3s;
  -webkit-transition: transform 0.3s,opacity 0.3s;
  transition: transform 0.3s,opacity 0.3s;
  -webkit-transition-timing-function: cubic-bezier(0.75,0,0.125,1);
  transition-timing-function: cubic-bezier(0.75,0,0.125,1);
}

.c0::before {
  opacity: 1;
  -webkit-transform: scale3d(1,1,1);
  -ms-transform: scale3d(1,1,1);
  transform: scale3d(1,1,1);
}

@media (min-width:20em) {
  .c0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5em) {
  .c0::before {
    border: 2px solid #FF6633;
    border-radius: 5px;
  }
}

@media (min-width:37.5em) {
  .c0 {
    border-radius: 5px;
    width: 7rem;
    width: -webkit-max-content;
    width: -moz-max-content;
    width: max-content;
  }
}

<a
  className="c0"
  data-isactive="/"
  href="/"
  onClick={[Function]}
/>
`;
